x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -265,3 +248,31 @@ func (k msgServer) updateSupplier(
 
 	return nil
 }
+
+// transferSupplierStakeDiff transfers the difference between the current and new stake
+// amounts by either escrowing when the stake is increased or unescrowing otherwise.
+func (k msgServer) transferSupplierStakeDiff(
+	ctx context.Context,
+	msgSignerAccAddress sdk.AccAddress,
+	supplierCurrentStake sdk.Coin,
+	newStake sdk.Coin,
+) error {
+	// The Supplier is increasing its stake, so escrow the difference
+	if supplierCurrentStake.Amount.LT(newStake.Amount) {
+		coinsToEscrow := sdk.NewCoins(newStake.Sub(supplierCurrentStake))
+
+		// Send the coins from the message signer account to the staked supplier pool
+		return k.bankKeeper.SendCoinsFromAccountToModule(ctx, msgSignerAccAddress, suppliertypes.ModuleName, coinsToEscrow)
+	}
+
+	// The supplier is decreasing its stake, unescrow the difference but deduct the staking fee.
```

Comment:
#PUC why we no newStake is greater than the minStake.

Optionally, consider being redundant with this logic. Doesn't hurt.